src/address.rs

```diff
@@ -51,57 +99,48 @@ impl<T, U> Address<T, U> {
 
 impl<T, U> Address<T, U>
 where
-    Address<usize, U>: From<Address<T, U>>,
-    Address<T, U>: Copy,
+    Self: Into<Address<usize, U>>,
 {
     /// Returns a raw pointer to its inner type
     ///
     /// # Safety
     /// Behavior is undefined, if the pointer is used and
     /// is not aligned or points to uninitialized memory.
-    pub fn as_ptr(&self) -> *const U {
-        Address::<usize, U>::from(*self).0 as *const U
+    #[inline]
+    pub fn as_ptr(self) -> *const U {
+        self.into().0 as *const U
     }
 
     /// Returns a raw pointer to its inner type
     ///
     /// # Safety
     /// Behavior is undefined, if the pointer is used and
     /// is not aligned or points to uninitialized memory.
-    pub fn as_mut_ptr(&self) -> *mut U {
-        Address::<usize, U>::from(*self).0 as *mut U
-    }
-}
-
-impl<T: Copy, U> fmt::Pointer for Address<T, U>
-where
-    Address<usize, U>: From<Address<T, U>>,
-    Address<T, U>: Copy,
-{
-    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        write!(f, "{:p}", self.as_ptr())
+    #[inline]
+    pub fn as_mut(self) -> *mut U {
```

Comment:
```suggestion
    pub fn as_mut_ptr(self) -> *mut U {
```

Reply:
I made that change intentionally for greater consistency with other types (since we're already breaking API). Rust generally has `as_mut()` or `as_ptr_mut()` but I haven't seen `as_mut_ptr()`.

Reply:
err... https://doc.rust-lang.org/std/primitive.slice.html#method.as_mut_ptr ... and MaybeUninit ... and, and, and

Reply:
@haraldh Fixed.